### PR TITLE
Fixes #9: remote_subnet is required for dynamic tunnel

### LINKED ISSUE #9: remote_subnet is required for dynamic tunnel
Even when set up to use dynamic, requires `remote_subnet`, otherwise returns:
```bash
$ terraform apply

Error: module.vpn-module-dynamic.google_compute_route.route: 1 error(s) occurred:

* module.vpn-module-dynamic.google_compute_route.route: __builtin_IntMath: divide by zero in:

${element(var.remote_subnet, (count.index%length(var.remote_subnet)))}
```
Configuration:
```terraform
resource "google_compute_router" "vpn_cloud_router" {
  name    = "vpn-cloud-router"
  region  = "${var.region}"
  network = "${var.network}"
  project = "${var.project}"

  bgp {
    asn = "64515"
  }
}

module "vpn-module-dynamic" {
  source = "terraform-google-modules/vpn/google"

  project_id         = "${var.project}"
  network            = "${var.network}"
  region             = "${var.region}"
  gateway_name       = "vpn-gw-dynamic"
  tunnel_name_prefix = "vpn-tn-dynamic"

  shared_secret = "secrets"
  tunnel_count  = 1
  peer_ips      = ["1.1.1.1"]

  cr_name = "${google_compute_router.vpn_cloud_router.name}"

  bgp_cr_session_range     = ["169.254.0.1/30", "169.254.0.3/30"]
  bgp_remote_session_range = ["169.254.0.2", "169.254.0.4"]
  peer_asn                 = ["64516", "64517"]

  remote_subnet = ["10.200.10.0/24", "10.200.20.0/24"]
}
```

### PR DESCRIPTION
**Pull request by nick4fake**
https://github.com/terraform-google-modules/terraform-google-vpn/issues/9

Added support for dynamic router name

### CODE REVIEW

File: tunnel.tf [hunk 1]
```diff
@@ -41,7 +41,7 @@ resource "google_compute_vpn_tunnel" "tunnel-static" {
 }
 
 resource "google_compute_vpn_tunnel" "tunnel-dynamic" {
-  count         = var.cr_name != "" ? var.tunnel_count : 0
+  count         = var.cr_enabled ? var.tunnel_count : 0
```
Comment (morgante) [inline tunnel.tf]: @nick4fake This logic looks like it will cause a change in behavior. Previously it's the inverse of the static block.

Comment (ingwarr) [reply tunnel.tf]: Previously, there was reliance on a non-empty name, now boolean variable `cr_enabled` was introduced, and if it is true, the `count` is not zero. And vice versa in the case of a static tunnel. In general, the logic is the same

**Comment by ingwarr** [comment]
Integration tests can be ignored due to https://github.com/terraform-google-modules/terraform-google-vpn/issues/15

File: tunnel.tf (hunk 1 above)
Comment (nick4fake) [reply tunnel.tf]: @ingwarr could you please add more info of why have you done this change?

File: examples/multi_tunnels/prod.tf [hunk 2]
```diff
@@ -49,6 +49,7 @@ module "vpn-gw-us-ce1-prd-mgt-internal-01" {
   peer_ips           = [module.vpn-gw-us-ce1-mgt-prd-internal-01.gateway_ip, module.vpn-gw-us-ce1-mgt-prd-internal-02.gateway_ip]
 
   cr_name                  = "cr-uscentral1-to-mgt-vpc-tunnels-01"
+  cr_enabled = true
```
Comment (ivankorn) [inline examples/multi_tunnels/prod.tf]: Doesn't brake formatting, but aligned might look better

File: examples/multi_tunnels/mgmt.tf [hunk 3]
```diff
@@ -65,7 +66,8 @@ module "vpn-gw-us-ce1-mgt-prd-internal-02" {
   tunnel_count       = 2
```
Comment (ivankorn) [inline examples/multi_tunnels/mgmt.tf]: Doesn't brake formatting, but aligned might look better

File: examples/multi_tunnels/mgmt.tf [hunk 4]
```diff
@@ -65,7 +66,8 @@ module "vpn-gw-us-ce1-mgt-prd-internal-02" {
...
   peer_ips           = [module.vpn-gw-us-ce1-prd-mgt-internal-02.gateway_ip, module.vpn-gw-us-ce1-prd-mgt-internal-01.gateway_ip]
```
Comment (ivankorn) [inline examples/multi_tunnels/mgmt.tf]: same

File: examples/multi_tunnels/mgmt.tf [hunk 5]
```diff
@@ -65,7 +66,8 @@ module "vpn-gw-us-ce1-mgt-prd-internal-02" {
...
 
-  cr_name                  = "cr-uscentral1-to-prod-vpc-tunnels-02"
+  cr_name    = "cr-uscentral1-to-prod-vpc-tunnels-02"
```
Comment (ivankorn) [inline examples/multi_tunnels/mgmt.tf]: same

File: examples/multi_tunnels/prod.tf [hunk 6]
```diff
@@ -66,6 +67,7 @@ module "vpn-gw-us-ce1-prd-mgt-internal-02" {
   peer_ips           = [module.vpn-gw-us-ce1-mgt-prd-internal-02.gateway_ip, module.vpn-gw-us-ce1-mgt-prd-internal-01.gateway_ip]
 
   cr_name                  = "cr-uscentral1-to-mgt-vpc-tunnels-02"
+  cr_enabled               = true
```
Comment (ivankorn) [inline examples/multi_tunnels/prod.tf]: Doesn't brake formatting, but aligned might look better

File: examples/multi_tunnels/prod.tf (hunk 2 above)
Comment (kopachevsky) [reply examples/multi_tunnels/prod.tf]: how come terraform fmt passed that?

Comment (ivankorn) [reply examples/multi_tunnels/prod.tf]: @kopachevsky, 1 white space is ok for it. 
So it passes the check, but still produces some level of distraction if the styles are mixed in 1 file.. I'm ok with the other way, but we may want to auto-format everything that way.

**Comment by nick4fake** [comment]
image

File: tunnel.tf (hunk 1 above)
Comment (nick4fake) [reply tunnel.tf]: @morgante I don't see good workarounds

Comment (nick4fake) [reply tunnel.tf]: cr_enabled seems as seems an equivalent for cr_name != ""

Comment (morgante) [reply tunnel.tf]: Ok

File: variables.tf [hunk 7]
```diff
@@ -92,9 +92,15 @@ default     = 1000
...
 variable "cr_name" {
-type        = string
-description = "The name of cloud router for BGP routing"
-default     = ""
+  type        = string
+  description = "The name of cloud router for BGP routing"
+  default     = ""
+}
+
+variable "cr_enabled" {
```
[the remark this answers is not in the document]
Comment (nick4fake) [reply variables.tf]: image
I can not

**Pull request merged**